### PR TITLE
Sort changelog, omit github link

### PR DESCRIPTION
**Pull request by xprazak2**
Sort the categories in changelog alphabetically by category name, add an option to generate changelog without links to github commits.

### CODE REVIEW

File: lib/tool_belt/changelog.rb [hunk 1]
```diff
@@ -35,9 +36,9 @@ def generate_entry(issue)
 
       list_item = title_string
       list_item += issue_string
-      list_item += ", "
 
-      list_item += hash_string(issue)
+      list_item += ", "
+      list_item += hash_string(issue) unless @omit_gh_link
```
Comment (akofink) [inline lib/tool_belt/changelog.rb]: No link also means no revision SHA? Seems like we would just want to omit the link part (the parenthesis).

File: lib/tool_belt/commands/changelog_command.rb [hunk 2]
```diff
@@ -21,7 +22,7 @@ def execute
           config.options.namespace,
           systools
         )
-        ToolBelt::Changelog.new(config.options, release_environment)
+        ToolBelt::Changelog.new(config.options, release_environment, short?)
```
Comment (akofink) [inline lib/tool_belt/commands/changelog_command.rb]: Why not use `omit_gh_link?`? It's much more descriptive as a variable and a cli option.

**Comment by xprazak2** [comment]
I kept the SHA and renamed the option to `omit-gh-link`

**Comment by ehelms** [comment]
@xprazak2 I don't mind the option being here, but it would help if you expand on what use case someone would want to omit the GH link to the commit?

**Comment by xprazak2** [comment]
What gets created here is almost the same as the release notes page on [theforeman.org](https://github.com/theforeman/theforeman.org/blob/gh-pages/_includes/manuals/1.16/1.2_release_notes.md), minus the links. And when I needed a list of changes for 1.17, I removed the links to have the same look.

I found out later that we have [a script](https://github.com/theforeman/theforeman.org/blob/gh-pages/scripts/release_notes.rb) to generate release notes. So I guess this is not really needed.

Closing, thanks!

**Comment by ehelms** [comment]
@xprazak2 Sad a little bit. I'd hope we could eliminate scripts to center on this one tool so we generate commonly structured release notes and changelogs.

**Comment by xprazak2** [comment]
Maybe having the links to a GH commit in our release log may not be a bad idea. I will revisit this when generating a changelog next time. I would also like to eliminate theforeman.org/scripts or at leas move the useful code here.

**Pull request closed without merging**